### ISSUE TITLE
upgrade 1.19.1 - lost of metric - consul.dns.domain_query

### ISSUE THREAD

**Issue by Spuxy** [hashicorp/consul#21574]
#### Overview of the Issue

Based on consul's upstream [doc](https://developer.hashicorp.com/consul/docs/agent/monitor/telemetry#metrics-reference), there is mention of metric `consul.dns.domain_query` but it seems that disappears with new dns version

#### Reproduction Steps

1. Cluster with 3 masters and 20 clients
2. 20 clients have ver. `1.18.0`
3. all communicate through protocol `2`
4. every server has status `alive`
5. based on this https://developer.hashicorp.com/consul/docs/upgrading/instructions/general-process#perform-the-upgrade
6. create snapshot
7. replace old binary with the new one (1.18.0 -> 1.19.1)
8. i did one by one non-leaders, the last was leader
9. all client have still ver. `1.18.0`, but endpoint `/v1/agent/metrics` doesnt return the query `consul.dns.domain_query`.
10. when i put `experiments: ["v1dns"]`, then it return the metric

### Consul info for both Client and Server

```
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease =
	revision = 9f62fb41
	version = 1.19.1
	version_metadata =
consul:
	acl = enabled
	bootstrap = false
	known_datacenters = 1
	leader = false
	leader_addr = xx
	server = true
raft:
	applied_index = 8249610
	commit_index = 8249610
	fsm_pending = 0
	last_contact = 37.083169ms
	last_log_index = 8249610
	last_log_term = 80
	last_snapshot_index = 8238376
	last_snapshot_term = 78
	latest_configuration = xx
	latest_configuration_index = 0
	num_peers = 2
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Follower
	term = 80
runtime:
	arch = amd64
	cpu_count = 4
	goroutines = 538
	max_procs = 4
	os = linux
	version = go1.22.5
serf_lan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 73
	failed = 1
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 87916
	members = 102
	query_queue = 0
	query_time = 2
serf_wan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 48
	members = 3
	query_queue = 0
```

```
Node               Address               Status  Type    Build   Protocol  DC     Partition  Segment
master1            1.1.1.1               alive   server  1.19.1  2         y      default    <all>
master2            1.1.1.1               alive   server  1.19.1  2         y      default    <all>
master3            1.1.1.1               alive   server  1.19.1  2         y      default    <all>
xx1                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
xx2                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
xx3                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
xx4                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
xx5                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
xx6                1.1.1.1               alive   client  1.18.0  2         y      default    <default>
....
```

the doc of metrics should be refactored, because if `v2` is new default for stable `1.19.0`, then it becomes irrelevant

**Comment by Spuxy** [comment]
Okay the answer will be this line https://github.com/hashicorp/consul/blob/main/agent/dns.go#L512 but still the doc needs to be changed

**Comment by jmurret** [comment]
This is resolved with https://github.com/hashicorp/consul/pull/21598 which reverts the experimental v2 dns and will be available in 1.19.2.
